cpp/examples/snap_dataset_to_graphar.cc

```diff
@@ -0,0 +1,134 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <fstream>
+#include <iostream>
+
+#include "./config.h"
+#include "gar/api.h"
+#include "gar/writer/edges_builder.h"
+#include "gar/writer/vertices_builder.h"
+
+// using facebook_combined.txt from SNAP dataset
+// available at https://snap.stanford.edu/data/ego-Facebook.html
+
+/*------------------original dataset status------------------*/
+#define GRAPH_NAME "facebook"
+#define DATA_PATH "/tmp/snap/original_dataset/facebook/facebook_combined.txt"
+#define VERTEX_COUNT 4039
+#define IS_DIRECTED false
+/*-----------------------GraphAr status---------------------*/
+#define SAVE_PATH "/tmp/snap/" + graph_name + "/"
+#define ADJLIST_TYPE GAR_NAMESPACE::AdjListType::ordered_by_source
+#define PAYLOAD_TYPE GAR_NAMESPACE::FileType::CSV
+#define VERTEX_CHUNK_SIZE 1024
+#define EDGE_CHUNK_SIZE 1024 * 1024
+
+int main(int argc, char* argv[]) {
+  std::string graph_name = GRAPH_NAME;
+  std::string save_path = SAVE_PATH;
+
+  /*------------------construct vertex info------------------*/
+  auto version = GAR_NAMESPACE::InfoVersion::Parse("gar/v1").value();
+
+  // meta info
+  std::string vertex_label = "node", vertex_prefix = "vertex/node/";
+
+  // create vertex info
+  auto vertex_info = GAR_NAMESPACE::CreateVertexInfo(
+      vertex_label, VERTEX_CHUNK_SIZE, {}, vertex_prefix, version);
+
+  // save & dump
+  ASSERT(!vertex_info->Dump().has_error());
+  ASSERT(vertex_info->Save(save_path + "node.vertex.yml").ok());
+
+  /*------------------construct edge info------------------*/
+  std::string src_label = "node", edge_label = "links", dst_label = "node",
+              edge_prefix = "edge/node_links_node/";
+  bool directed = IS_DIRECTED;
+
+  // construct adjacent lists
+  auto adjacent_lists = {
+      GAR_NAMESPACE::CreateAdjacentList(ADJLIST_TYPE, PAYLOAD_TYPE)};
+  // create edge info
+  auto edge_info = GAR_NAMESPACE::CreateEdgeInfo(
+      src_label, edge_label, dst_label, EDGE_CHUNK_SIZE, VERTEX_CHUNK_SIZE,
+      VERTEX_CHUNK_SIZE, directed, adjacent_lists, {}, edge_prefix, version);
+
+  // save & dump
+  ASSERT(!edge_info->Dump().has_error());
+  ASSERT(edge_info->Save(save_path + "node_links_node.edge.yml").ok());
+
+  /*------------------construct graph info------------------*/
+  // create graph info
+  auto graph_info = GAR_NAMESPACE::CreateGraphInfo(
+      graph_name, {vertex_info}, {edge_info}, save_path, version);
+  // save & dump
+  ASSERT(!graph_info->Dump().has_error());
+  ASSERT(graph_info->Save(save_path + graph_name + ".graph.yml").ok());
+
+  /*------------------construct vertices------------------*/
+  // construct vertices builder
+  GAR_NAMESPACE::IdType start_index = 0;
+  GAR_NAMESPACE::builder::VerticesBuilder v_builder(vertex_info, save_path,
```

Comment:
I suggest we should use the `Make` method to create builder to avoid raise exception dircectly.
https://github.com/alibaba/GraphAr/blob/fc91672e191f8bf3ae79302da793ca4db38eb6ec/cpp/include/gar/writer/edges_builder.h#L284-L322